services/metadactyl-clj/src/metadactyl/routes/apps.clj

```diff
@@ -102,79 +102,79 @@
         :summary "Update an App"
         :description "This service updates a single-step App in the database, as long as the App has not
         been submitted for public use."
-        (service/trap uri apps/update-app current-user (assoc body :id app-id)))
+        (ok (apps/update-app current-user (assoc body :id app-id))))
 
-  (POST* "/:app-id/copy" [:as {uri :uri}]
+  (POST* "/:app-id/copy" []
          :path-params [app-id :- AppIdPathParam]
          :query [params SecuredQueryParamsRequired]
          :return App
          :summary "Make a Copy of an App Available for Editing"
          :description "This service can be used to make a copy of an App in the user's workspace."
-         (service/trap uri apps/copy-app current-user app-id))
+         (ok (apps/copy-app current-user app-id)))
 
-  (GET* "/:app-id/details" [:as {uri :uri}]
+  (GET* "/:app-id/details" []
         :path-params [app-id :- AppIdJobViewPathParam]
         :query [params SecuredQueryParams]
         :return AppDetails
         :summary "Get App Details"
         :description "This service is used by the DE to obtain high-level details about a single App"
-        (service/coerced-trap uri AppDetails apps/get-app-details current-user app-id))
+        (service/coerced-trap nil AppDetails apps/get-app-details current-user app-id))
```

Comment:
I think either way is fine. Probably a matter of preference, whether `(service/coerce Schema func args...)` or `(mc/coerce! Schema (func args...))` suits your aesthetic preferences better.

I notice that `mc/coerce!` throws `ring.swagger.schema/validation` which is the semi-deprecated error code compojure-api used to throw for request/response validation, so that might also be worth updating (though the code I changed to catch the new errors still catches the old ones because of things like this).